server/lib/src/idm/account.rs

```diff
@@ -248,8 +246,10 @@ impl Account {
     where
         TXN: QueryServerTransaction<'a>,
     {
-        let ((groups, unix_groups), rap) =
-            load_all_groups_from_account_entry_with_policy(value, qs)?;
+        let groups = Group::<()>::try_from_account(value, qs)?;
+        let unix_groups: Vec<Group<Unix>> =
+            Group::<Unix>::try_from_account(value, qs).unwrap_or_default();
```

Comment:
Another double-iterate here.

Reply:
there are about 5 lying around. I'll fix that up with a single call that does both

Reply:
I wonder if the answer is something like we did before with account and account_unix_extn. Where we iterate over groups once and they optionally have unix extn. 

Saying that I do wonder if these "intermediate" types need to just be removed and re-thought somehow.

Reply:
I mean, The second part of this PR is giving the same treatment to the Account. We want the properties to be known at compile time rather than runtime. The fact we have to check `unix_extn` every time is what is killing me. We should know during compilation what properties this struct has.